source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+
+      $ sudo apt-get update
+
+#. Now that you have the repository configured, it's time to upgrade
+   the ``cloudstack-management`` package.
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-usage
+
+
+.. _rhel412:
+.. _rpm-repo412:
+
+CentOS/RHEL
+##############
+
+If you are using CentOS or RHEL, follow this procedure to upgrade your
+packages. If not, skip to hypervisors section :ref:`upg_hyp_412`.
+
+.. note::
+   **Community Packages:** This section assumes you're using the community
+   supplied packages for CloudStack. If you've created your own packages and
+   yum repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the yum repository
+for each system with CloudStack packages. This means all
+management servers, and any hosts that have the KVM agent.
+
+(No changes should be necessary for hosts that are running VMware
+or Xen.)
+
+Start by opening ``/etc/yum.repos.d/cloudstack.repo`` on any
+systems that have CloudStack packages installed.
+
+This file should have content similar to the following:
+
+.. parsed-literal::
+
+   [apache-cloudstack]
+   name=Apache CloudStack
+   baseurl=http://download.cloudstack.org/centos/6/4.12/
+   enabled=1
+   gpgcheck=0
+
+If you are using the community provided package repository, change
+the base url to:
+
+.. parsed-literal::
+
+   http://download.cloudstack.org/centos/$releasever/|version|/
+
+Setup the GPG public key if you wish to enable ``gpgcheck=1``:
+
+.. parsed-literal::
+
+   rpm --import http://download.cloudstack.org/RPM-GPG-KEY
+
+
+
+If you're using your own package repository, change this line to
+read as appropriate for your |version| repository.
+
+#. Now that you have the repository configured, it's time to upgrade the
+   ``cloudstack-management``.
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-usage
+
+.. _upg_hyp_412:
+
+Upgrade Hypervisors
+-------------------
+
+Hypervisor: XenServer
+#####################
+
+
+No additional steps are required wrt for XenServer Hypervisor for this upgrade.
+
+
+Hypervisor: VMware
+###################
+
+.. warning::
+   For VMware hypervisor CloudStack management server packages must be
+   build using "noredist". Refer to :ref:`building-noredist`.
+
+
+No additional steps are requried for the VMware Hypervisor for this upgrade.
+
+
+.. _kvm412:
+
+Hypervisor: KVM
+#################
+
+KVM on Ubuntu
+""""""""""""""
+
+(KVM only) Additional steps are required for each KVM host. These
+steps will not affect running guests in the cloud. These steps are
+required only for clouds using KVM as hosts and only on the KVM
+hosts.
+
+#. Configure the :ref:`APT repo <apt-repo411>` as detailed above.
```

Comment:
apt-repo413

Reply:
I'm not sure how they're used, it's been from before I did not change it. Will need review/checking. It's also in the previous 4.11, 4.12 rst files. cc @PaulAngus

Reply:
fixed